src/screens/SearchScreen.js

```diff
@@ -33,8 +33,8 @@ class SearchScreen extends Component {
     componentWillReceiveProps(nextProps){
         const {sitesShouldUpdate, lastKnownRegion} = nextProps;
 
-        if(sitesShouldUpdate){
-            this.props.initializeMap(lastKnownRegion);
+        if(sitesShouldUpdate && sitesShouldUpdate !== this.props.sitesShouldUpdate){
+            this.props.initializeMap({region: lastKnownRegion});
```

Comment:
This was the fix for the infinite loop I found. Similar to what I was saying in one of the other PRs earlier last week, if we have `componentWillReceiveProps` fire off an action, that action (which will very likely trigger a reducer) will then trigger new props, which then triggers a new `componentWillReceiveProps` call. For that reason, we have to be really careful about when we use this to trigger actions. The way I worked around this one was to only fire off again if sitesShouldUpdate is `true` *AND* if the last known props and new props are different.